packages/preview-middleware-client/test/unit/adp/controllers/AddFragment.controller.test.ts

```diff
@@ -0,0 +1,324 @@
+
+            expect(valueStateSpy.mock.calls.length).toBe(1);
+        });
+
+        test('sets error when the fragment name is has special characters', () => {
+            const addFragment = new AddFragment('adp.extension.controllers.AddFragment');
+
+            const valueStateSpy = jest.fn();
+            const event = {
+                getSource: jest.fn().mockReturnValue({
+                    getValue: jest.fn().mockReturnValue('Share 2$5!'),
+                    setValueState: valueStateSpy,
+                    setValueStateText: jest.fn()
+                })
+            };
+
+            addFragment.model = {
+                setProperty: jest.fn(),
+                getProperty: jest.fn().mockReturnValue([{ fragmentName: 'Delete.fragment.xml' }])
+            } as unknown as JSONModel;
+
+            addFragment.dialog = {
+                getBeginButton: jest.fn().mockReturnValue({ setEnabled: jest.fn() })
+            } as unknown as Dialog;
+
+            addFragment.onFragmentNameInputChange(event as unknown as Event);
+
+            expect(valueStateSpy.mock.calls.length).toBe(1);
+        });
+
+        test('sets create button to true when the fragment name is valid', () => {
+            const addFragment = new AddFragment('adp.extension.controllers.AddFragment');
+
+            const valueStateSpy = jest.fn();
+            const event = {
+                getSource: jest.fn().mockReturnValue({
+                    getValue: jest.fn().mockReturnValue('Share'),
+                    setValueState: valueStateSpy,
+                    setValueStateText: jest.fn()
+                })
+            };
+
+            addFragment.model = {
+                setProperty: jest.fn(),
+                getProperty: jest.fn().mockReturnValue([{ fragmentName: 'Delete.fragment.xml' }])
+            } as unknown as JSONModel;
+
+            addFragment.dialog = {
+                getBeginButton: jest.fn().mockReturnValue({ setEnabled: jest.fn() })
+            } as unknown as Dialog;
+
+            addFragment.onFragmentNameInputChange(event as unknown as Event);
+
+            expect(valueStateSpy.mock.calls.length).toBe(1);
+        });
+    });
+
+    describe('onCreateBtnPress', () => {
+        afterEach(() => {
+            jest.restoreAllMocks();
+        });
+
+        test('creates new fragment', async () => {
+            const addFragment = new AddFragment('adp.extension.controllers.AddFragment');
+
+            const event = {
+                getSource: jest.fn().mockReturnValue({
+                    setEnabled: jest.fn()
+                })
+            };
+
+            addFragment.model = {
+                getProperty: jest
+                    .fn()
+                    .mockReturnValueOnce('Share')
+                    .mockReturnValueOnce('0')
+                    .mockReturnValueOnce('content')
+            } as unknown as JSONModel;
+
+            const createFragmentSpy = jest.fn();
+            // @ts-ignore
```

Comment:
You shouldn't use `@ts-ignore` here, instead think about how to test without ts complaining.

The main issue here is that your tests assume the existence of a private function which is internal knowledge of the unit you are testing, so your test should not rely on that. Instead think about what the actual result of this test case should be.
If I read the code correctly then in a positive case it should be a call to the public backend API which for the client-side code is the call to `fetch`. So, instead of mocking internal knowledge, mock the call to fetch that is required to create a new fragment. At the end of the test you validate that the correct data is being sent to the backend (with fetch), don't just check if fetch is called.

Once you did this, you can also delete you test of private functions below/

Reply:
`@ts-ignore` is gone, thank you, however, I am not sure I like that you made private functions public to get rid of it. The test for `onCreateBtnPress` should cover everything you split into the tests for `createNewFragment` and `createFragmentChange`

Reply:
Made the methods private and reworked tests in: https://github.com/SAP/open-ux-tools/pull/1183/commits/1e3e4f2d1f48d5a510e8ad6dfe37f105cb2b0cd9